scrapy_zyte_api/providers.py

```
@@ -91,6 +92,9 @@ async def __call__(  # noqa: C901
         for cls in to_provide:
             cls_stripped = strip_annotated(cls)
             assert isinstance(cls_stripped, type)
+            if cls_stripped is Geolocation and is_typing_annotated(cls):
+                zyte_api_meta["geolocation"] = cls.__metadata__[0]  # type: ignore[attr-defined]
```

Should we raise an error / warning if it's not annotated?